lib/pbench/server/filetree.py

```diff
@@ -298,18 +312,34 @@ def unpack(self, incoming: Path, results: Path):
         ensured by calling this indirectly through the Controller class unpack
         method.
 
-        TODO: This is a prototype for testing, and doesn't actually unpack
-        the tarball as we're going to be relying on the current unpack pipeline
-        command for some time.
-
         Args:
             incoming: Controller's directory in the INCOMING tree
             results: Controller's directory in the RESULTS tree
+
+        Raises:
+            TarballUnpackError if an exception/failure occurs while unpacking
+            the tarball.
         """
-        self.logger.warning("Tarball unpack is not yet implemented")
         unpacked = incoming / self.name
-        unpacked.mkdir(parents=True)  # Just create an empty directory for now
+        unpacked.mkdir(parents=True)
         self.unpacked = unpacked
```

Comment:
Looking at Peter's unresolved conversation ... there was a suggestion made that setting `self.unpacked` could be deferred until *after* the `try/else` sequence so that we set it only when the unpack has been successful, and *not* if we're raising `TarballUnpackError`.

Even though I just approved 😆 🍤 , I think this change would be worth making, and might also resolve Peter's concerns. (The actual behavior following that error is really up to the caller, in particular `pbench-unpack-tarballs`, which is literally a matter for a separate PR.)